CONTRIBUTING.md

```diff
@@ -1,55 +1,69 @@
+## Building
 
-To build the `main` branch of SwiftSyntax, follow the following instructions:
-
-1. Check `swift-syntax` and  `swift` out side by side:
-    ```
-    - (enclosing directory)
-      - swift
-      - swift-syntax
-    ```
+You can either build the project by opening it in Xcode or by executing the following command on the command line:
+```bash
+swift-syntax/build-script.py build --toolchain /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-<recent date>.xctoolchain/usr
+```
 
-2. Make sure you have a recent [Trunk Swift Toolchain](https://swift.org/download/#snapshots) installed.
-3. Define the `TOOLCHAINS` environment variable as below to have the `swift` command point inside the toolchain:
+## Testing
 
-    ```bash
-    $ export TOOLCHAINS=swift
-    ```
+Because of SwiftSyntax’s integration with the Swift compiler project, testing certain parts of the project is a little bit more involved than others. 
 
-4. To make sure everything is set up correctly, check the return statement of `xcrun --find swift`. It should point inside the latest installed trunk development toolchain. If it points inside an Xcode toolchain, check that you exported the `TOOLCHAINS` environment variable correctly. If it points inside a version-specific toolchain (like Swift 5.0-dev), you'll need to remove that toolchain.
-5. Run `swift-syntax/build-script.py build --toolchain /path/to/recent/swift/development/snapshot.xctoolchain/usr`.
-    If despite following those instructions, you get compiler errors, the Swift toolchain might be too old to contain recent changes in Swift's SwiftSyntaxParser C library. In that case, you'll have to build the compiler and SwiftSyntax together with the following command:
+The `SwiftParser` tests have no depdendencies, other `XCTests` require you to use a matching Swift Development Snapshot, `lit`-based test require a compiler build.
 
-    ```bash
-    $ swift/utils/build-script --swiftsyntax --swiftpm --llbuild
-    ```
+Run the tests that you belive are necessary locally. CI will always run all test before allowing the PR to be merged.
 
-Swift-CI will automatically run the code generation step whenever a new toolchain (development snapshot or release) is published. It should thus almost never be necessary to perform the above build yourself.
+### SwiftParser
 
-Afterward, SwiftPM can also generate an Xcode project to develop SwiftSyntax by running `swift package generate-xcodeproj`.
+If you are only modifying the `SwiftParser` module, you can just run the tests using Xcode by testing the `SwiftParserTest` target. 
 
-If you also want to run tests locally, read the section below as testing has additional requirements.
+If you can’t find it in your Schemes, you need to manually add it using Product -> Scheme -> New Scheme…
```

Comment:
This shouldn't be necessary in a package. The products we declare in the Package.swift file should always appear as build targets in the Xcode UI. To run the tests, you can select the appropriate product and then

Product > Test